[sonic test] introduce a repeat harness

yxieca opened the pull request:
### Type of change
- [x ] Testbed and Framework(new/improvement)

### Approach
How did you do it?
Introduce a test harness that repeats a user defined task list n times.

By default, n is 10, user can override n with
  -extra-vars 'test_iterations=<num>'

How did you verify/test it?
Use repeat harness to execute reboot test.

lguohan commented:
have you tried the run test by name, I think it will be easier to add repeat parameter there.

yxieca commented:
Thanks, I added following changes:

1. removed test by tag access to repeat harness.
2. added repeat to the test by name infrastructure.
3. left the repeat_harness in test by name infrastructure for repeating a list of steps.
